### DISCUSSION TITLE
Same Graph through different GCN layers in same times.

### DISCUSSION THREAD

**Discussion by jiaruHithub** [pyg-team/pytorch_geometric#2195] [Q&A]
Hi!
I have a questuion about GCN layer.
I want to make same graph enter two GCN layers (same layers, but different weights).
Like this:
```
                                Graph
                       /                \
                     /                   \
                   GCN1               GCN3
                    |                    |
                   GCN2               GCN4
```
I write the code as fellow:
```
        x1, x2 = F.relu(self.conv1(x, edge_index)), F.relu(self.conv3(x, edge_index))
        x1, x2 = F.dropout(x1, p=args.dropout, training=self.training), F.dropout(x2, p=args.dropout, training=self.training)
        x1, x2 = self.conv2(x1, edge_index), self.conv4(x2, edge_index)
```
But it seems don't work. In training, The training time is even less than a single consecutive layer.
How can I write this code and make it work?

**Comment by wsad1** [comment]
I don't see any issues with your `forward` function. Also training time could be lesser for other reasons, and that need not mean the code isn't working.

**Reply by rusty1s** [reply]
Indeed, the code looks correct. Can you give us some more details? What do you mean by "it does not work"?

**Reply by jiaruHithub** [reply]
I m sorry for my expressive approach. My meaning is that the training speed of two channels is less than one channel. Another thing is the accuracy of two methods is completely same.

**Reply by rusty1s** [reply]
Can you give a minimal example of what you are doing concretely? This might make it easier to help you :)
